src/db.c

```diff
@@ -650,47 +651,44 @@ long long emptyData(int dbnum, int flags, void(callback)(hashtable *)) {
     return removed;
 }
 
-/* Initialize temporary db on replica for use during diskless replication. */
-serverDb *initTempDb(void) {
-    int slot_count_bits = 0;
-    int flags = KVSTORE_ALLOCATE_HASHTABLES_ON_DEMAND;
-    if (server.cluster_enabled) {
-        slot_count_bits = CLUSTER_SLOT_MASK_BITS;
-        flags |= KVSTORE_FREE_EMPTY_HASHTABLES;
-    }
-    serverDb *tempDb = zcalloc(sizeof(serverDb) * server.dbnum);
-    for (int i = 0; i < server.dbnum; i++) {
-        tempDb[i].id = i;
-        tempDb[i].keys = kvstoreCreate(&kvstoreKeysHashtableType, slot_count_bits, flags);
-        tempDb[i].expires = kvstoreCreate(&kvstoreExpiresHashtableType, slot_count_bits, flags);
-    }
-
-    return tempDb;
-}
-
-/* Discard tempDb, it's always async. */
-void discardTempDb(serverDb *tempDb) {
+/* Empty and release all databases in the temp db array and set them to NULL. */
```

Comment:
Hm... :thinking: I suggested this comment but missed the `zfree(tempDb)` in the bottom of the function. :smile: So we do actually free the whole thing.

I guess I was confused by the fact that we set `tempDb[i] = NULL;` if we're going to free the array anyway. It's unnecessary to set the elements to null here but I guess it's considered good practice...

```suggestion
/* Discard tempDb array. It's always async. */
```